### ISSUE TITLE
shard a table using ```SERIAL``` column

### ISSUE THREAD

**Issue by forkeer** [citusdata/citus#1446]
Hi
I want to shard a table using SERIAL column , how can I do this

ERROR:
values given for the partition column must be constants or constant expressions
ERROR:
distributed modifications must target exactly one shard

**Comment by forkeer** [comment]
Sample Run Query 
```
 CREATE TABLE IF NOT EXISTS "user" (
  "id"                   SERIAL       NOT NULL,
  "status"               SMALLINT     NOT NULL DEFAULT 0,
  "nickname"             VARCHAR(255) NOT NULL DEFAULT '',
  "gender"               SMALLINT     NOT NULL DEFAULT 0,
  "email"                VARCHAR(255) NOT NULL DEFAULT '',
);
```


```
SELECT master_create_distributed_table('user', 'id', 'hash');
SELECT master_create_worker_shards('user', 16, 1);
```

`
INSERT INTO "user" ("status","nickname","gender") VALUES (1,'John',1) RETURNING "id";

`

**Comment by onderkalaci** [comment]
This works on Citus 6.2 (see  an example from the [tests](https://github.com/citusdata/citus/blob/a7c94cb36d60ec2970b70794abb9020ce0a4b09d/src/test/regress/expected/multi_modifications.out#L599) which looks almost the same with your example). Could you make sure that you're on 6.2 by executing `\dx` on psql?

**Comment by forkeer** [comment]
Thanks Yes 
This issue solved in 6.2
